sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs

```diff
@@ -500,9 +504,11 @@ private sealed class PipelineResponse : Response
             private Stream? _contentStream;
 #pragma warning restore CA2213
 
-            public PipelineResponse(string requestId, HttpResponseMessage responseMessage, Stream? contentStream)
+            public PipelineResponse(string requestId, HttpResponseMessage responseMessage, Stream? contentStream, ClientOptions options, ResponseClassifier classifier)
             {
                 ClientRequestId = requestId ?? throw new ArgumentNullException(nameof(requestId));
+                ExceptionFactory = new ResponseExceptionFactory(options);
```

Comment:
We probably don't want an allocation here?  I think I can refactor to achieve that.